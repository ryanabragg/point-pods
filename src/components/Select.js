// adjusted from https://material-ui.com/demos/autocomplete/#react-select

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Select from 'react-select';
import Creatable from 'react-select/lib/Creatable';

import { withStyles } from '@material-ui/core/styles';
import { emphasize } from '@material-ui/core/styles/colorManipulator';

import NoSsr from '@material-ui/core/NoSsr';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import Chip from '@material-ui/core/Chip';
import MenuItem from '@material-ui/core/MenuItem';

const styles = theme => ({
  root: {
    flexGrow: 1,
    height: 250,
  },
  input: {
    display: 'flex',
    padding: 0,
  },
  valueContainer: {
    display: 'flex',
    flexWrap: 'wrap',
    flex: 1,
    alignItems: 'center',
  },
  chip: {
    margin: `${theme.spacing.unit / 2}px ${theme.spacing.unit / 4}px`,
  },
  chipFocused: {
    backgroundColor: emphasize(
      theme.palette.type === 'light' ? theme.palette.grey[300] : theme.palette.grey[700],
      0.08,
    ),
  },
  noOptionsMessage: {
    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 2}px`,
  },
  singleValue: {
    fontSize: 16,
  },
  placeholder: {
    position: 'absolute',
    left: 2,
    fontSize: 16,
  },
  paper: {
    marginTop: theme.spacing.unit,
  },
  divider: {
    height: theme.spacing.unit * 2,
  },
});

const NoOptionsMessage = (props) => {
  return (
    <Typography
      color='textSecondary'
      className={props.selectProps.classes.noOptionsMessage}
      {...props.innerProps}
    >
      {props.children}
    </Typography>
  );
}

const inputComponent = ({ inputRef, ...props }) => {
  return <div ref={inputRef} {...props} />;
}

const Control = (props) => {
  return (
    <TextField
      fullWidth
      InputProps={{
        inputComponent,
        inputProps: {
          className: props.selectProps.classes.input,
          inputRef: props.innerRef,
          children: props.children,
          ...props.innerProps,
        },
      }}
      {...props.selectProps.textFieldProps}
    />
  );
}

const Option = (props) => {
  return (
    <MenuItem
      buttonRef={props.innerRef}
      selected={props.isFocused}
      disabled={props.isDisabled}
      component='div'
      style={{
        fontWeight: props.isSelected ? 500 : 400,
      }}
      {...props.innerProps}
    >
      {props.children}
    </MenuItem>
  );
}

const Placeholder = (props) => {
  return (
    <Typography
      color='textSecondary'
      className={props.selectProps.classes.placeholder}
      {...props.innerProps}
    >
      {props.children}
    </Typography>
  );
}

const SingleValue = (props) => {
  return (
    <Typography className={props.selectProps.classes.singleValue} {...props.innerProps}>
      {props.children}
    </Typography>
  );
}

const ValueContainer = (props) => {
  return <div className={props.selectProps.classes.valueContainer}>{props.children}</div>;
}

const MultiValue = (props) => {
  return (
    <Chip
      tabIndex={-1}
      label={props.children}
      className={classNames(props.selectProps.classes.chip, {
        [props.selectProps.classes.chipFocused]: props.isFocused,
      })}
      onDelete={event => {
        props.removeProps.onClick();
        props.removeProps.onMouseDown(event);
      }}
    />
  );
}

const Menu = (props) => {
  return (
    <Paper square className={props.selectProps.classes.paper} {...props.innerProps}>
      {props.children}
    </Paper>
  );
}

const components = {
  Option,
  Control,
  NoOptionsMessage,
  Placeholder,
  SingleValue,
  MultiValue,
  ValueContainer,
  //Menu,
};

class IntegrationReactSelect extends React.Component {
  render() {
    const { classes, theme, isCreatable, label, ...rest } = this.props;

    const selectStyles = {
      input: base => ({
        ...base,
        color: theme.palette.text.primary,
      }),
    };

    const textFieldProps = {
      label: label,
      InputLabelProps: {
        shrink: true,
      },
    };

    return (
      <NoSsr>
        {isCreatable
          ? (
            <Creatable
              classes={classes}
              styles={selectStyles}
              textFieldProps={textFieldProps}
              components={components}
              {...rest}
            />
          )
          : (
            <Select
              classes={classes}
              styles={selectStyles}
              textFieldProps={textFieldProps}
              components={components}
              {...rest}
            />
          )
        }
      </NoSsr>
    );
  }
}

IntegrationReactSelect.defaultProps = {
  isCreatable: false,
  label: '',
};

IntegrationReactSelect.propTypes = {
  classes: PropTypes.object.isRequired, // added by withStyles
  theme: PropTypes.object.isRequired, // added by withStyles
  isCreatable: PropTypes.bool,
  label: PropTypes.string,
  // Select and Creatable props
};

export default withStyles(styles, { withTheme: true })(IntegrationReactSelect);
