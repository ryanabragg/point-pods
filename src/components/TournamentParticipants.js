import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import Toolbar from '@material-ui/core/Toolbar';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import Button from '@material-ui/core/Button';

import AddIcon from '@material-ui/icons/Add';
import ClearIcon from '@material-ui/icons/Clear';

import IntegrationAutosuggest from '../components/IntegrationAutosuggest';

import { withAPI } from '../api';

const styles = theme => ({
  right: {
    marginLeft: 'auto',
  },
  addPlayer: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'nowrap',
    alignItems: 'center',
    justifyContent: 'center',
  },
  list: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    '& > ul': {
      width: '100%',
      maxWidth: 420,
    },
    margin: theme.spacing.unit,
    bottom: 0,
  },
});

class TournamentParticipants extends Component {
  state = {
    menuAnchor: null,
    tournamentID: null,
    players: [],
    participants: [],
  };

  componentDidMount() {
    if(!this.props.match.params.id)
      return this.props.history.push('/');
    let loadState = { tournamentID: this.props.match.params.id };
    this.props.api.Tournaments.get(this.props.match.params.id)
      .catch(error => {
        console.log(error.message);
        this.props.history.push('/');
      })
      .then(tournament => loadState.participants = tournament.players)
      .then(() => this.props.api.Players.all())
      .then(list => loadState.players = list)
      .catch(error => console.log(error.message))
      .then(() => this.setState(loadState));
  }

  handleToggleMenu = (show) => event => {
    const value = event ? event.currentTarget : null;
    this.setState({ menuAnchor: show ? value : null });
  };

  handleSelect = ({ id, label }) => {
    if(!label.length)
      return;
    return new Promise((resolve, reject) => {
      const found = this.state.players.filter(player => player.id === id);
      if(id === null || !found.length)
        return reject('Player record not found');
      resolve(found[0]);
    })
      .catch(error => this.props.api.Players.set({name: label}))
      .then(player => {
        this.setState({ players: this.state.players.concat(player) });
        console.log('info', `Created player ${player.name}`, player);
        return player;
      })
      .then(player => {
        const added = this.state.participants.filter(p => p.id === player.id);
        if(added.length)
          throw new Error('Player already added');
        console.log('info', `Added player ${player.name}`, player);
        return this.state.participants.concat(player).map(player => ({
          id: player.id,
          name: player.name,
          points: 0,
        })).sort((a, b) => {
          if(a.name < b.name)
            return -1;
          else if(a.name > b.name)
            return 1;
          return 0;
        });
      })
      .then(participants => {
        return this.props.api.Tournaments.set({
          id: this.state.tournamentID,
          players: participants,
        })
        .then(tournament => {
          this.setState({ participants: participants });
        });
      })
      .catch(error => console.log('error', error.message));
  };

  removeParticipant = event => {
    const id = event.currentTarget.id;
    const participants = this.state.participants.filter(p => String(p.id) !== id);
    this.props.api.Tournaments.set({
      id: this.state.tournamentID,
      players: participants,
    })
      .then(() => {
        this.setState({
          participants: participants,
        });
      });
  };

  removeAllParticipants = event => {
    this.props.api.Tournaments.set({
      id: this.state.tournamentID,
      players: [],
    })
      .then(() => {
        this.setState({
          menuAnchor: null,
          participants: [],
        });
      });
  };

  render() {
    const { classes } = this.props;
    const { menuAnchor, players, participants} = this.state;
    return (
      <div>
        <Toolbar>
          <Typography variant='headline'>
            Add Players
          </Typography>
          <Typography>
          </Typography>
          <Button className={classes.right}
            aria-label='Players Menu'
            aria-owns={menuAnchor ? 'players-menu' : null}
            aria-haspopup='true'
            onClick={this.handleToggleMenu(true)}
          >
            {`${participants.length} Player${participants.length === 1 ? '' : 's'}`}
          </Button>
          <Menu
            id='players-menu'
            anchorEl={menuAnchor}
            open={Boolean(menuAnchor)}
            onClose={this.handleToggleMenu(false)}
          >
            <MenuItem onClick={this.removeAllParticipants}>Remove All</MenuItem>
          </Menu>
        </Toolbar>
        <div className={classes.addPlayer}>
          <IntegrationAutosuggest
            items={players
              .filter(player => -1 === participants.findIndex(p => p.id === player.id))
              .map(p => ({ id: p.id, label: p.name }))
            }
            onSelect={this.handleSelect}
            clearOnSelect={true}
            buttonIcon={<AddIcon />}
          />
        </div>
        <div className={classes.list}>
          <List>
            {participants.map(player => (
              <ListItem key={player.id}>
                <IconButton aria-label='Remove Player' id={player.id} onClick={this.removeParticipant}>
                  <ClearIcon />
                </IconButton>
                <ListItemText primary={player.name} secondary={`Points: ${player.points}`}/>
              </ListItem>
            ))}
          </List>
        </div>
      </div>
    );
  }
}

TournamentParticipants.defaultProps = {
  history: {
    push: () => null,
    replace: () => null,
  },
  match: {
    params: {
      id: null,
    },
  },
};

TournamentParticipants.propTypes = {
  api: PropTypes.object.isRequired, // added by withAPI
  classes: PropTypes.object.isRequired, // added by withStyles
  history: PropTypes.object, // added by parent
  match: PropTypes.object, // added by parent
};

export default withStyles(styles, { withTheme: true })(withAPI(TournamentParticipants));
