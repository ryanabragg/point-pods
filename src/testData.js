import api from './api';

const testSettings = {
  pairingMethod: api.pairingMethods.HISTORIC,
  pairingMethodInitial: api.pairingMethods.HISTORIC,
  podSizeMinimum: 4,
  podSizeMaximum: 7,
};

const testPlayers = [{
  id: 'mock0',
  name: 'test0',
  points: 0,
}, {
  id: 'mock1',
  name: 'test1',
  points: 1,
}, {
  id: 'mock2',
  name: 'test2',
  points: 2,
}, {
  id: 'mock3',
  name: 'test3',
  points: 3,
}, {
  id: 'mock4',
  name: 'test4',
  points: 4,
}, {
  id: 'mock5',
  name: 'test5',
  points: 5,
}, {
  id: 'mock6',
  name: 'test6',
  points: 6,
}, {
  id: 'mock7',
  name: 'test7',
  points: 7,
}, {
  id: 'mock8',
  name: 'test8',
  points: 8,
}, {
  id: 'mock9',
  name: 'test9',
  points: 9,
}];

const testTournaments = [{
  id: 'mock0',
  name: 'test0',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Etiam urna tortor, laoreet vitae commodo porttitor, mattis id purus.',
  date: new Date('2018-01-01'),
  done: false,
  staging: true,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 3,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 3 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 0,
    participated: false,
    dropped: false,
  }],
}, {
  id: 'mock1',
  name: 'test1',
  category: 'cat1',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed molestie suscipit suscipit. Nullam sollicitudin tempus pretium.',
  date: new Date('2018-02-01'),
  done: true,
  staging: false,
  pairingMethod: api.pairingMethods.POINTS,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 5,
  rounds: 2,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 0 },
    2: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 1 },
    2: { pod: 2, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
    2: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 3 },
    2: { pod: 2, points: 3 },
  }, {
    id: 'mock4',
    name: 'test4',
    points: 4,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 4 },
    2: { pod: 1, points: 4 },
  }, {
    id: 'mock5',
    name: 'test5',
    points: 5,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 5 },
    2: { pod: 2, points: 5 },
  }, {
    id: 'mock6',
    name: 'test6',
    points: 6,
    participated: true,
    dropped: false,
    2: { pod: 2, points: 6 },
  }],
}, {
  id: 'mock2',
  name: 'test2',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Sed ullamcorper metus vitae nisi rhoncus placerat.',
  date: new Date('2018-01-01'),
  done: false,
  staging: false,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 3 },
  }],
}];

const mockRoute = {
  history: {
    push: jest.fn(to => to),
    replace: jest.fn(to => to),
  },
  match: {
    params: {},
  },
  matchWithInvalid: {
    params: {
      id: 'invalid',
    },
  },
  matchWithID0: {
    params: {
      id: 'mock0',
    },
  },
  matchWithID1: {
    params: {
      id: 'mock1',
    },
  },
};

let mockStore = {};
const mockStorage = {
  setItem: jest.fn((key, val) => Promise.resolve((mockStore[key] = val))),
  getItem: jest.fn(key => Promise.resolve(mockStore[key])),
  removeItem: jest.fn(key => { delete mockStore[key]; Promise.resolve(); }),
  clear: jest.fn(() => Promise.resolve((mockStore = {}))),
  keys: jest.fn(() => Promise.resolve(Object.keys(mockStore))),
  length: jest.fn(() => Promise.resolve(Object.keys(mockStore).length)),
};

let defaultStore = null;
function mockAPI(api, storage) {
  defaultStore = api.store;
  api.store = storage;
  mockStore = {
    settings: testSettings,
    players: testPlayers,
    tournaments: testTournaments,
  };
}
function demockAPI(api) {
  api.store = defaultStore;
  defaultStore = null;
}

export {
  testSettings,
  testPlayers,
  testTournaments,
  mockRoute,
  mockStore,
  mockStorage,
  mockAPI,
  demockAPI,
};
