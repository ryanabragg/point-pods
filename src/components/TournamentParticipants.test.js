import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import AppBar from '@material-ui/core/AppBar';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import IconButton from '@material-ui/core/IconButton';

import TournamentParticipants from './TournamentParticipants';
import Select from './Select';

import { testPlayers, testTournaments } from '../testData';

const playerSort = jest.fn((a, b) => {
  if(a.name < b.name)
    return -1;
  else if(a.name > b.name)
    return 1;
  return 0;
});

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TournamentParticipants />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', () => {
  const component = renderer.create(<TournamentParticipants />);
  expect(component.toJSON()).toMatchSnapshot();
  component.update(
    <TournamentParticipants
      players={testTournaments[0].players}
      allPlayers={testPlayers}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
  component.update(
    <TournamentParticipants
      players={testTournaments[0].players}
      allPlayers={testPlayers}
      sort={playerSort}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
  component.update(
    <TournamentParticipants
      players={testTournaments[0].players}
      allPlayers={testPlayers}
      sort={playerSort}
      displayOnly={true}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

describe('actions', () => {
  test('adding a player', () => {
    const spy = jest.fn();
    const component = shallow(
      <TournamentParticipants
        players={testTournaments[0].players}
        allPlayers={testPlayers}
        onSelectPlayer={spy}
      />
    ).dive();
    expect(component.find(Select).prop('onChange')).toEqual(component.instance().handleSelect);
    component.instance().handleSelect(testPlayers[8], 'test');
    expect(spy.mock.calls[0][0]).toBe(testPlayers[8]);
    expect(spy.mock.calls[0][1]).toBe('test');
    component.setProps({ displayOnly: true });
    expect(component.find(Select).length).toBe(0);
  });

  test('adding a new player', () => {
    const spy = jest.fn();
    const component = shallow(
      <TournamentParticipants
        players={testTournaments[0].players}
        allPlayers={testPlayers}
        onSelectCreatePlayer={spy}
      />
    ).dive();
    expect(component.find(Select).prop('onCreateOption')).toEqual(component.instance().handleCreate);
    component.instance().handleCreate('testing');
    expect(spy.mock.calls[0][0]).toBe('testing');
    component.setProps({ displayOnly: true });
    expect(component.find(Select).length).toBe(0);
  });

  test('removing a player (non-participating)', () => {
    const spy = jest.fn();
    const component = shallow(
      <TournamentParticipants
        players={testTournaments[0].players}
        allPlayers={testPlayers}
        onRemovePlayer={spy}
      />
    ).dive();
    component.find(ListItem).at(3).find(IconButton).simulate('click');
    expect(spy.mock.calls[0][0]).toBe(testTournaments[0].players[3].id);
    component.setProps({ displayOnly: true });
    expect(component.find(IconButton).length).toBe(0);
  });

  test('removing a player (drop participant)', () => {
    const spy = jest.fn();
    const component = shallow(
      <TournamentParticipants
        players={testTournaments[0].players}
        allPlayers={testPlayers}
        onRemovePlayer={spy}
      />
    ).dive();
    component.find(ListItem).at(2).find(IconButton).simulate('click');
    expect(spy.mock.calls[0][0]).toBe(testTournaments[0].players[2].id);
    component.setProps({ displayOnly: true });
    expect(component.find(IconButton).length).toBe(0);
  });

  test('reinstating a player (un-drop)', () => {
    const spy = jest.fn();
    const component = shallow(
      <TournamentParticipants
        players={testTournaments[0].players}
        allPlayers={testPlayers}
        onReinstatePlayer={spy}
      />
    ).dive();
    component.find(ListItem).at(0).find(IconButton).simulate('click');
    expect(spy.mock.calls[0][0]).toBe(testTournaments[0].players[0].id);
    component.setProps({ displayOnly: true });
    expect(component.find(IconButton).length).toBe(0);
  });
});

test('displayOnly', () => {
  const component = shallow(
    <TournamentParticipants
      players={testTournaments[0].players}
      allPlayers={testPlayers}
    />
  ).dive();
  expect(component.find(AppBar).length).toBe(1);
  expect(component.find(IconButton).length).toBe(4);
  component.setProps({ displayOnly: true });
  expect(component.find(AppBar).length).toBe(0);
  expect(component.find(IconButton).length).toBe(0);
});

test('printing', () => {
  const component = shallow(
    <TournamentParticipants
      players={testTournaments[0].players}
      allPlayers={testPlayers}
    />
  ).dive();
  expect(component.find(AppBar).prop('className')).toBe('TournamentParticipants-hideOnPrint-98');
  expect(component.find(List).at(0).prop('className')).toBe('TournamentParticipants-hideOnPrint-98');
  expect(component.find('div').at(1).prop('className')).toBe('TournamentParticipants-showOnPrint-99');
});
