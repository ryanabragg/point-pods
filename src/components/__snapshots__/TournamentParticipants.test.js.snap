// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the prior snapshot 1`] = `
<div>
  <div
    className="MuiToolbar-root-139 MuiToolbar-regular-141 MuiToolbar-gutters-140"
  >
    <h1
      className="MuiTypography-root-143 MuiTypography-headline-148"
    >
      Add Players
    </h1>
    <p
      className="MuiTypography-root-143 MuiTypography-body1-152"
    />
    <button
      aria-haspopup="true"
      aria-label="Players Menu"
      aria-owns={null}
      className="MuiButtonBase-root-191 MuiButton-root-167 MuiButton-text-169 MuiButton-flat-172 APIComponent-right-136"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-168"
      >
        0 Players
      </span>
      <span
        className="MuiTouchRipple-root-194"
      />
    </button>
  </div>
  <div
    className="APIComponent-addPlayer-137"
  >
    <div
      className="IntegrationAutosuggest-root-205"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="react-autowhatever-1"
        className="IntegrationAutosuggest-container-206"
        role="combobox"
      >
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-210 MuiFormControl-fullWidth-213"
        >
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-1"
            className="MuiInput-root-214 MuiInput-fullWidth-221 MuiInput-formControl-215 MuiInput-underline-218"
            style={Object {}}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={undefined}
              className="MuiInput-input-222"
              defaultValue={undefined}
              disabled={false}
              id={undefined}
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={undefined}
              placeholder="Add a player"
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiPaper-root-227 MuiPaper-elevation2-231"
          id="react-autowhatever-1"
          role="listbox"
          style={Object {}}
        />
      </div>
      <button
        className="MuiButtonBase-root-191 MuiIconButton-root-254 MuiIconButton-colorInherit-255"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-259"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-260"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-194"
        />
      </button>
    </div>
  </div>
  <div
    className="APIComponent-list-138"
  >
    <ul
      className="MuiList-root-267 MuiList-padding-268"
    />
  </div>
</div>
`;

exports[`matches the prior snapshot 2`] = `
<div>
  <div
    className="MuiToolbar-root-139 MuiToolbar-regular-141 MuiToolbar-gutters-140"
  >
    <h1
      className="MuiTypography-root-143 MuiTypography-headline-148"
    >
      Add Players
    </h1>
    <p
      className="MuiTypography-root-143 MuiTypography-body1-152"
    />
    <button
      aria-haspopup="true"
      aria-label="Players Menu"
      aria-owns={null}
      className="MuiButtonBase-root-191 MuiButton-root-167 MuiButton-text-169 MuiButton-flat-172 APIComponent-right-136"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-168"
      >
        0 Players
      </span>
      <span
        className="MuiTouchRipple-root-194"
      />
    </button>
  </div>
  <div
    className="APIComponent-addPlayer-137"
  >
    <div
      className="IntegrationAutosuggest-root-205"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="react-autowhatever-1"
        className="IntegrationAutosuggest-container-206"
        role="combobox"
      >
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-210 MuiFormControl-fullWidth-213"
        >
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls="react-autowhatever-1"
            className="MuiInput-root-214 MuiInput-fullWidth-221 MuiInput-formControl-215 MuiInput-underline-218"
            style={Object {}}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={undefined}
              className="MuiInput-input-222"
              defaultValue={undefined}
              disabled={false}
              id={undefined}
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={undefined}
              placeholder="Add a player"
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiPaper-root-227 MuiPaper-elevation2-231"
          id="react-autowhatever-1"
          role="listbox"
          style={Object {}}
        />
      </div>
      <button
        className="MuiButtonBase-root-191 MuiIconButton-root-254 MuiIconButton-colorInherit-255"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-259"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-260"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-194"
        />
      </button>
    </div>
  </div>
  <div
    className="APIComponent-list-138"
  >
    <ul
      className="MuiList-root-267 MuiList-padding-268"
    />
  </div>
</div>
`;
